### ISSUE TITLE
Dont work with gulp4

### ISSUE THREAD

**Issue by perryutkonos** [twolfson/gulp.spritesmith#135]
## Dont work gulp.spritesmith in gulp4.
About in gulp 3 its ok g
gulp.spritesmith version - 6.7.0

```
TypeError: file.pipe is not a function
    at loadPixels (D:\Projects\Web\PROJ\local\.dev\node_modules\pixelsmith\lib\engine.js:40:16)
    at fn (D:\Projects\Web\PROJ\local\.dev\node_modules\pixelsmith\node_modules\async\lib\async.js:638:34)
    at Immediate.<anonymous> (D:\Projects\Web\PROJ\local\.dev\node_modules\pixelsmith\node_modules\async\lib\async.js:554:34)
    at runCallback (timers.js:672:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
```
Gulp task:
```gulp.task('sprite', function () {

var spriteData = gulp.src(path.src.sprite)
    .pipe(spritesmith({
    imgName: '../img/sprite.png',
    cssName: 'sprite.less',
    padding: 20
}));

// Pipe image stream through image optimizer and onto disk
var imgStream = spriteData.img
// DEV: We must buffer our stream into a Buffer for `imagemin`
    .pipe(gulp.dest(path.build.img));

// Pipe CSS stream through CSS optimizer and onto disk
var cssStream = spriteData.css
    .pipe(gulp.dest('src/css/blocks/'));

// Return a merged stream to handle both `end` events
return merge(imgStream, cssStream);
});
```

**Comment by twolfson** [comment]
This has been an open issue on `pixelsmith` for a while (https://github.com/twolfson/pixelsmith/issues/10). However, I haven't seen/heard anything about a Gulp@4 release soon. Could you point me to this information? I don't even see Gulp@4 on npm =/

```
$ npm show gulp
{ name: 'gulp',
  description: 'The streaming build system',
  'dist-tags': { latest: '3.9.1' },
  versions:
   [ '0.0.1',
// ...
     '3.9.0',
     '3.9.1' ],
  maintainers:
```

**Comment by perryutkonos** [comment]
I'm using alpha version gulp4 (https://github.com/gulpjs/gulp/tree/4.0), but I did not have problems with all Npm modules, not counting the spritesmith ((

**Comment by perryutkonos** [comment]
@twolfson Is there a solution for using spritesmithn with a version of gulp4?

**Comment by twolfson** [comment]
In my opinion, Gulp@4 is still in development and thus a moving target so it's a waste of energy to try to support/use it at the moment. When there's a beta release on `npm` or in their `git tag`, then we should revisit it but until then there's nothing saying it's stable or near release =/

**Comment by bm-stschneider** [comment]
The problem seems to be relatively simple. in `engine.js` of the pixelsmith package in line 37 is a isNull check. It seems this is currently returning an incorrect value since although there is a file with path and so on isNull returns false.
So if you really want to use Gulp 4 you could change this or comment out the if condition and then it works.
But as @twolfson already said, gulp 4 being in alpha makes it difficult to fix this.

**Comment by aslinwang** [comment]
gulp 4 has been released.
https://github.com/gulpjs/gulp/tree/v4.0.0

**Comment by twolfson** [comment]
Cool, thanks for the heads up! I'll re-open this issue. Please see https://github.com/twolfson/pixelsmith/issues/10 for ETA and resolution

**Comment by twolfson** [comment]
This has been resolved and released in `6.9.0`
